diff -ru libcore/clone.rs libcore_patched/clone.rs
--- libcore/clone.rs	2015-07-02 13:02:27.891621659 +0800
+++ libcore_patched/clone.rs	2015-07-02 13:02:04.780163524 +0800
@@ -80,7 +80,9 @@
 clone_impl! { u32 }
 clone_impl! { u64 }
 
+#[cfg(not(disable_float))]
 clone_impl! { f32 }
+#[cfg(not(disable_float))]
 clone_impl! { f64 }
 
 clone_impl! { () }
diff -ru libcore/default.rs libcore_patched/default.rs
--- libcore/default.rs	2015-07-02 13:02:27.887621753 +0800
+++ libcore_patched/default.rs	2015-07-02 13:02:04.784163432 +0800
@@ -153,5 +153,7 @@
 default_impl! { i32, 0 }
 default_impl! { i64, 0 }
 
+#[cfg(not(disable_float))]
 default_impl! { f32, 0.0f32 }
+#[cfg(not(disable_float))]
 default_impl! { f64, 0.0f64 }
diff -ru libcore/fmt/mod.rs libcore_patched/fmt/mod.rs
--- libcore/fmt/mod.rs	2015-07-02 13:02:27.895621566 +0800
+++ libcore_patched/fmt/mod.rs	2015-07-02 13:02:04.784163432 +0800
@@ -17,6 +17,7 @@ use prelude::*;
 use cell::{Cell, RefCell, Ref, RefMut, BorrowState};
 use marker::PhantomData;
 use mem;
+#[cfg(not(disable_float))]
 use num::flt2dec;
 use ops::Deref;
 use result;
@@ -991,6 +654,7 @@ impl<'a> Formatter<'a> {
     /// Takes the formatted parts and applies the padding.
     /// Assumes that the caller already has rendered the parts with required precision,
     /// so that `self.precision` can be ignored.
+    #[cfg(not(disable_float))]
     fn pad_formatted_parts(&mut self, formatted: &flt2dec::Formatted) -> Result {
         if let Some(mut width) = self.width {
             // for the sign-aware zero padding, we render the sign first and
@@ -1027,6 +691,7 @@ impl<'a> Formatter<'a> {
         }
     }
 
+    #[cfg(not(disable_float))]
     fn write_formatted_parts(&mut self, formatted: &flt2dec::Formatted) -> Result {
         fn write_bytes(buf: &mut Write, s: &[u8]) -> Result {
             buf.write_str(unsafe { str::from_utf8_unchecked(s) })
@@ -1391,6 +1057,7 @@ impl<'a, T> Pointer for &'a mut T {
     }
 }
 
+#[cfg(not(disable_float))]
 // Common code of floating point Debug and Display.
 fn float_to_decimal_common<T>(fmt: &mut Formatter, num: &T, negative_zero: bool) -> Result
     where T: flt2dec::DecodableFloat
@@ -1415,6 +1082,7 @@ fn float_to_decimal_common<T>(fmt: &mut Formatter, num: &T, negative_zero: bool)
     fmt.pad_formatted_parts(&formatted)
 }
 
+#[cfg(not(disable_float))]
 // Common code of floating point LowerExp and UpperExp.
 fn float_to_exponential_common<T>(fmt: &mut Formatter, num: &T, upper: bool) -> Result
     where T: flt2dec::DecodableFloat
@@ -1468,7 +1136,9 @@ macro_rules! floating { ($ty:ident) => {
         }
     }
 } }
+#[cfg(not(disable_float))]
 floating! { f32 }
+#[cfg(not(disable_float))]
 floating! { f64 }
 
 // Implementation of Display/Debug for various core types
diff -ru libcore/intrinsics.rs libcore_patched/intrinsics.rs
--- libcore/intrinsics.rs	2015-07-02 13:02:27.891621659 +0800
+++ libcore_patched/intrinsics.rs	2015-07-02 13:02:04.780163524 +0800
@@ -393,7 +392,10 @@
     pub fn volatile_load<T>(src: *const T) -> T;
     /// Perform a volatile store to the `dst` pointer.
     pub fn volatile_store<T>(dst: *mut T, val: T);
+}
 
+#[cfg(not(disable_float))]
+extern "rust-intrinsic" {
     /// Returns the square root of an `f32`
     pub fn sqrtf32(x: f32) -> f32;
     /// Returns the square root of an `f64`
@@ -490,7 +492,9 @@
     pub fn roundf32(x: f32) -> f32;
     /// Returns the nearest integer to an `f64`. Rounds half-way cases away from zero.
     pub fn roundf64(x: f64) -> f64;
+}
 
+extern "rust-intrinsic" {
     /// Returns the number of bits set in a `u8`.
     pub fn ctpop8(x: u8) -> u8;
     /// Returns the number of bits set in a `u16`.
diff -ru libcore/lib.rs libcore_patched/lib.rs
--- libcore/lib.rs	2015-07-02 13:02:27.899621472 +0800
+++ libcore_patched/lib.rs	2015-07-02 13:02:04.780163524 +0800
@@ -114,7 +110,9 @@
 #[path = "num/u32.rs"]  pub mod u32;
 #[path = "num/u64.rs"]  pub mod u64;
 
+#[cfg(not(disable_float))]
 #[path = "num/f32.rs"]   pub mod f32;
+#[cfg(not(disable_float))]
 #[path = "num/f64.rs"]   pub mod f64;
 
 #[macro_use]
@@ -160,2 +150,3 @@
 #[cfg(not(stage0))]
+#[cfg(not(disable_float))]
 pub mod simd;
diff -ru libcore/num/flt2dec/decoder.rs libcore_patched/num/flt2dec/decoder.rs
--- libcore/num/flt2dec/decoder.rs	2015-07-02 13:02:27.903621378 +0800
+++ libcore_patched/num/flt2dec/decoder.rs	2015-07-02 13:02:04.784163432 +0800
@@ -12,6 +12,7 @@
 
 use prelude::v1::*;
 
+#[cfg(not(disable_float))]
 use {f32, f64};
 use num::{Float, FpCategory};
 
@@ -60,6 +61,8 @@
+#[cfg(not(disable_float))]
 impl DecodableFloat for f32 {
     fn min_pos_norm_value() -> Self { f32::MIN_POSITIVE }
 }
 
+#[cfg(not(disable_float))]
 impl DecodableFloat for f64 {
     fn min_pos_norm_value() -> Self { f64::MIN_POSITIVE }
diff -ru libcore/num/mod.rs libcore_patched/num/mod.rs
--- libcore/num/mod.rs	2015-07-02 13:02:27.903621378 +0800
+++ libcore_patched/num/mod.rs	2015-07-02 13:02:04.784163432 +0800
@@ -22,1 +22,1 @@
-use marker::{Copy, Sized};
+use marker::{Copy};
@@ -46,2 +46,4 @@
+#[cfg(not(disable_float))]
 pub mod flt2dec;
+#[cfg(not(disable_float))]
 pub mod dec2flt;
@@ -98,2 +93,3 @@
 }
+#[cfg(not(disable_float))]
 zero_one_impl_float! { f32 f64 }
@@ -1269,1 +1271,2 @@
+#[cfg(not(disable_float))]
 pub trait Float: Sized {
@@ -1472,1 +1473,2 @@
+#[cfg(not(disable_float))]
 pub use num::dec2flt::ParseFloatError;
diff -ru libcore/ops.rs libcore_patched/ops.rs
--- libcore/ops.rs	2015-07-02 13:02:27.899621472 +0800
+++ libcore_patched/ops.rs	2015-07-02 13:02:04.780163524 +0800
@@ -203,7 +203,9 @@
     )*)
 }
 
-add_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 f32 f64 }
+add_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 }
+#[cfg(not(disable_float))]
+add_impl! { f32 f64 }
 
 /// The `Sub` trait is used to specify the functionality of `-`.
 ///
@@ -257,7 +259,9 @@
     )*)
 }
 
-sub_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 f32 f64 }
+sub_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 }
+#[cfg(not(disable_float))]
+sub_impl! { f32 f64 }
 
 /// The `Mul` trait is used to specify the functionality of `*`.
 ///
@@ -311,7 +315,9 @@
     )*)
 }
 
-mul_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 f32 f64 }
+mul_impl! { usize u8 u16 u32 u64 isize i8 i16 i32 i64 }
+#[cfg(not(disable_float))]
+mul_impl! { f32 f64 }
 
 /// The `Div` trait is used to specify the functionality of `/`.
 ///
@@ -383,7 +371,8 @@
     )*)
 }
 
-div_impl_float! { f32 f64 }
+#[cfg(not(disable_float))]
+div_impl_float! { f32 f64 }
 
 /// The `Rem` trait is used to specify the functionality of `%`.
 ///
@@ -459,2 +466,3 @@
 #[cfg(not(stage0))]
+#[cfg(not(disable_float))]
 rem_impl_float! { f32 f64 }
@@ -566,7 +534,9 @@
 }
 
 // neg_impl_unsigned! { usize u8 u16 u32 u64 }
-neg_impl_numeric! { isize i8 i16 i32 i64 f32 f64 }
+neg_impl_numeric! { isize i8 i16 i32 i64 }
+#[cfg(not(disable_float))]
+neg_impl_numeric! { f32 f64 }
 
 /// The `Not` trait is used to specify the functionality of unary `!`.
 ///
