diff -rub libcore_orig/default.rs libcore/default.rs
--- libcore_orig/default.rs	2017-06-22 16:51:20.271895100 +0800
+++ libcore/default.rs	2017-06-22 16:51:09.831744196 +0800
@@ -157,2 +157,4 @@
+#[cfg(not(disable_float))]
 default_impl! { f32, 0.0f32, "Returns the default value of `0.0`" }
+#[cfg(not(disable_float))]
 default_impl! { f64, 0.0f64, "Returns the default value of `0.0`" }
diff -rub libcore_orig/fmt/mod.rs libcore/fmt/mod.rs
--- libcore_orig/fmt/mod.rs	2017-06-22 16:51:20.055891975 +0800
+++ libcore/fmt/mod.rs	2017-06-22 16:51:09.831744196 +0800
@@ -15,12 +15,14 @@
 use cell::{UnsafeCell, Cell, RefCell, Ref, RefMut};
 use marker::PhantomData;
 use mem;
+#[cfg(not(disable_float))]
 use num::flt2dec;
 use ops::Deref;
 use result;
 use slice;
 use str;
 
+#[cfg(not(disable_float))]
 mod float;
 mod num;
 mod builders;
@@ -1172,6 +1174,7 @@
     /// Takes the formatted parts and applies the padding.
     /// Assumes that the caller already has rendered the parts with required precision,
     /// so that `self.precision` can be ignored.
+    #[cfg(not(disable_float))]
     fn pad_formatted_parts(&mut self, formatted: &flt2dec::Formatted) -> Result {
         if let Some(mut width) = self.width {
             // for the sign-aware zero padding, we render the sign first and
@@ -1211,6 +1214,7 @@
         }
     }
 
+    #[cfg(not(disable_float))]
     fn write_formatted_parts(&mut self, formatted: &flt2dec::Formatted) -> Result {
         fn write_bytes(buf: &mut Write, s: &[u8]) -> Result {
             buf.write_str(unsafe { str::from_utf8_unchecked(s) })
diff -rub libcore_orig/intrinsics.rs libcore/intrinsics.rs
--- libcore_orig/intrinsics.rs	2017-06-22 16:51:20.215894290 +0800
+++ libcore/intrinsics.rs	2017-06-22 16:51:09.831744196 +0800
@@ -1080,7 +1080,10 @@
     /// The stabilized version of this intrinsic is
     /// [`std::ptr::write_volatile`](../../std/ptr/fn.write_volatile.html).
     pub fn volatile_store<T>(dst: *mut T, val: T);
+}
 
+#[cfg(not(disable_float))]
+extern "rust-intrinsic" {
     /// Returns the square root of an `f32`
     pub fn sqrtf32(x: f32) -> f32;
     /// Returns the square root of an `f64`
@@ -1197,8 +1200,9 @@
     /// Float remainder that allows optimizations based on algebraic rules.
     /// May assume inputs are finite.
     pub fn frem_fast<T>(a: T, b: T) -> T;
+}
 
-
+extern "rust-intrinsic" {
     /// Returns the number of bits set in an integer type `T`
     pub fn ctpop<T>(x: T) -> T;
 
Only in libcore/: intrinsics.rs.orig
diff -rub libcore_orig/iter/traits.rs libcore/iter/traits.rs
--- libcore_orig/iter/traits.rs	2017-06-22 16:51:20.255894869 +0800
+++ libcore/iter/traits.rs	2017-06-22 16:51:09.831744196 +0800
@@ -700,6 +700,7 @@
     );
 }
 
+#[cfg(not(disable_float))]
 macro_rules! float_sum_product {
     ($($a:ident)*) => ($(
         #[stable(feature = "iter_arith_traits", since = "1.12.0")]
@@ -733,6 +734,7 @@
 }
 
 integer_sum_product! { i8 i16 i32 i64 i128 isize u8 u16 u32 u64 u128 usize }
+#[cfg(not(disable_float))]
 float_sum_product! { f32 f64 }
 
 /// An iterator adapter that produces output as long as the underlying
diff -rub libcore_orig/lib.rs libcore/lib.rs
--- libcore_orig/lib.rs	2017-06-22 16:51:20.003891223 +0800
+++ libcore/lib.rs	2017-06-22 16:51:09.831744196 +0800
@@ -126,7 +126,9 @@
 #[path = "num/u64.rs"]   pub mod u64;
 #[path = "num/u128.rs"]   pub mod u128;
 
+#[cfg(not(disable_float))]
 #[path = "num/f32.rs"]   pub mod f32;
+#[cfg(not(disable_float))]
 #[path = "num/f64.rs"]   pub mod f64;
 
 #[macro_use]
diff -rub libcore_orig/num/flt2dec/decoder.rs libcore/num/flt2dec/decoder.rs
--- libcore_orig/num/flt2dec/decoder.rs	2017-06-22 16:51:20.187893885 +0800
+++ libcore/num/flt2dec/decoder.rs	2017-06-22 16:51:09.831744196 +0800
@@ -10,6 +10,7 @@
 
 //! Decodes a floating-point value into individual parts and error ranges.
 
+#[cfg(not(disable_float))]
 use {f32, f64};
 use num::FpCategory;
 use num::dec2flt::rawfp::RawFloat;
@@ -56,10 +57,12 @@
     fn min_pos_norm_value() -> Self;
 }
 
+#[cfg(not(disable_float))]
 impl DecodableFloat for f32 {
     fn min_pos_norm_value() -> Self { f32::MIN_POSITIVE }
 }
 
+#[cfg(not(disable_float))]
 impl DecodableFloat for f64 {
     fn min_pos_norm_value() -> Self { f64::MIN_POSITIVE }
 }
diff -rub libcore_orig/num/mod.rs libcore/num/mod.rs
--- libcore_orig/num/mod.rs	2017-06-22 16:51:20.043891802 +0800
+++ libcore/num/mod.rs	2017-06-22 16:51:09.831744196 +0800
@@ -90,7 +90,9 @@
 mod wrapping;
 
 // All these modules are technically private and only exposed for coretests:
+#[cfg(not(disable_float))]
 pub mod flt2dec;
+#[cfg(not(disable_float))]
 pub mod dec2flt;
 pub mod bignum;
 pub mod diy_float;
@@ -2404,6 +2406,7 @@
 #[unstable(feature = "core_float",
            reason = "stable interface is via `impl f{32,64}` in later crates",
            issue = "32110")]
+#[cfg(not(disable_float))]
 pub trait Float: Sized {
     /// Returns `true` if this value is NaN and false otherwise.
     #[stable(feature = "core", since = "1.6.0")]
@@ -2714,6 +2717,7 @@
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+#[cfg(not(disable_float))]
 pub use num::dec2flt::ParseFloatError;
 
 // Conversion traits for primitive integer and float types
@@ -2773,6 +2777,9 @@
 // they fit in the significand, which is 24 bits in f32 and 53 bits in f64.
 // Lossy float conversions are not implemented at this time.
 
+#[cfg(not(disable_float))]
+mod _int_flot_conv {
+use convert::From;
 // Signed -> Float
 impl_from! { i8, f32, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
 impl_from! { i8, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
@@ -2789,3 +2796,4 @@
 
 // Float -> Float
 impl_from! { f32, f64, #[stable(feature = "lossless_float_conv", since = "1.6.0")] }
+}
Only in libcore/num: mod.rs.orig
diff -rub libcore_orig/ops/arith.rs libcore/ops/arith.rs
--- libcore_orig/ops/arith.rs	2017-06-22 16:51:20.127893017 +0800
+++ libcore/ops/arith.rs	2017-06-22 16:51:09.831744196 +0800
@@ -116,7 +116,9 @@
     )*)
 }
 
-add_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+add_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+add_impl! { f32 f64 }
 
 /// The subtraction operator `-`.
 ///
@@ -190,7 +192,9 @@
     )*)
 }
 
-sub_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+sub_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+sub_impl! { f32 f64 }
 
 /// The multiplication operator `*`.
 ///
@@ -313,7 +317,9 @@
     )*)
 }
 
-mul_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+mul_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+mul_impl! { f32 f64 }
 
 /// The division operator `/`.
 ///
@@ -445,6 +451,7 @@
 
 div_impl_integer! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
 
+#[cfg(not(disable_float))]
 macro_rules! div_impl_float {
     ($($t:ty)*) => ($(
         #[stable(feature = "rust1", since = "1.0.0")]
@@ -459,6 +466,7 @@
     )*)
 }
 
+#[cfg(not(disable_float))]
 div_impl_float! { f32 f64 }
 
 /// The remainder operator `%`.
@@ -526,6 +534,7 @@
 rem_impl_integer! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
 
 
+#[cfg(not(disable_float))]
 macro_rules! rem_impl_float {
     ($($t:ty)*) => ($(
         #[stable(feature = "rust1", since = "1.0.0")]
@@ -540,6 +549,7 @@
     )*)
 }
 
+#[cfg(not(disable_float))]
 rem_impl_float! { f32 f64 }
 
 /// The unary negation operator `-`.
@@ -620,7 +630,9 @@
 }
 
 // neg_impl_unsigned! { usize u8 u16 u32 u64 }
-neg_impl_numeric! { isize i8 i16 i32 i64 i128 f32 f64 }
+neg_impl_numeric! { isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+neg_impl_numeric! { f32 f64 }
 
 /// The addition assignment operator `+=`.
 ///
@@ -677,7 +689,9 @@
     )+)
 }
 
-add_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+add_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+add_assign_impl! { f32 f64 }
 
 /// The subtraction assignment operator `-=`.
 ///
@@ -734,7 +748,9 @@
     )+)
 }
 
-sub_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+sub_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+sub_assign_impl! { f32 f64 }
 
 /// The multiplication assignment operator `*=`.
 ///
@@ -780,7 +796,9 @@
     )+)
 }
 
-mul_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+mul_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+mul_assign_impl! { f32 f64 }
 
 /// The division assignment operator `/=`.
 ///
@@ -825,7 +843,9 @@
     )+)
 }
 
-div_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+div_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+div_assign_impl! { f32 f64 }
 
 /// The remainder assignment operator `%=`.
 ///
@@ -870,4 +890,6 @@
     )+)
 }
 
-rem_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 f32 f64 }
+rem_assign_impl! { usize u8 u16 u32 u64 u128 isize i8 i16 i32 i64 i128 }
+#[cfg(not(disable_float))]
+rem_assign_impl! { f32 f64 }
Only in libcore/ops: arith.rs.orig
Only in libcore/ops: arith.rs.rej
Only in libcore_orig/: ops.rs
Only in libcore_orig/: ops.rs.rej
